### ISSUE TITLE
Add content around how BOLT12  `send_invoice` feature

### ISSUE THREAD

**Issue by Bosch-0** [BitcoinDesign/Guide#679]
With BOLT12 on the horizon, it has a new feature called `send_invoice` which allows a user to send an invoice to another user through the Lightning network rather than share it out of band (say over SMS or email).

How would this look like on the receiver of the requests end? How would they go from getting the request to making the payment?

**Comment by GBKS** [comment]
Could you find any documentation on this? I was not able to.

From a pure UX perspective, I think this falls into the general "[presenting payment requests](https://bitcoin.design/guide/payments/send/#presenting-payment-requests)" user flow. Whether the invoice got into the app via link click, QR code scan or otherwise is not so relevant to the presentation. Only addition might be a notification, and possibly a small indicator on the home screen showing that there's a pending invoice.

**Comment by Bosch-0** [comment]
Not really no, I was discussing this in the [bolt12 offer Telegram ](https://t.me/bolt12org)

There would be some slight UX differences during the [sharing payment requests](https://bitcoin.design/guide/payments/request/#sharing-payment-requests) on the requesting page also for this.

**Comment by GBKS** [comment]
What would be the UX difference? Another button to share directly via the Lightning network?

**Comment by Bosch-0** [comment]
> Another button to share directly via the Lightning network?

Most likely something like this yes. If you go and try out how Zebedee users request payments from other Zebedee users this is the UX I'm talking about.

**Comment by sbddesign** [comment]
> If you go and try out how Zebedee users request payments from other Zebedee users this is the UX I'm talking about.

So I agree that Zebedee is an example of a user sending another user a payment request. I would lop Zebedee, Strike, and CashApp together as examples of appswhere you can send payment requests to other users. That's definitely a distinct user flow.

However, I am not entirely certain that BOLT 12 `send_invoice` facilitates this use case. My understanding is that `send_invoice` is really in response to an offer. In other words, my lightning node would not just send an invoice to another node out of the blue -- it would send an invoice to that node only when prompted via a BOLT 12 offer.

I think we have to think of it in reverse. Imagine you have your friend's BOLT 12 offer stored in their contact metadata, then you can treat that as serving the same purpose as a CashTag, Strike userrname, or Zebedee gamer tag.

**Comment by Bosch-0** [comment]
> it would send an invoice to that node only when prompted via a BOLT 12 offer

I thought the same when I first was reading into this. I asked in the BOLT 12 offer telegram and was told that the above UX is possible. You are correct though, `send_invoice` is just how the invoice is shared over the network which is part of how this works but not the whole picture, I misunderstood that. I changed the OP and title of this issue to reflect that. 

I'm not 100% on the details of how this works, but with offers (very similar to LNURL-withdraw) pull payments are possible. Here is a scenario:

1) I'm a user who needs a refund from a merchant and they have sent me an offer
2) I can scan the offer, and rather then them serve me an invoice (how standard invoices are served) I can send them an invoice from my node that they pay.
3) I get my refund. 

What I wanted to address in this issue two things:

1) When the user scans the offer, if I can request (send them an invoice) or pay (receive an invoice) this needs to be reflected in the UI. 
 2) How on the merchants end they would receive this request. Would they just get a notification saying 'User X sent you an invoice' of which they would click and pay. This would most likely need to appear in the users activity. 

To complicate this further subscriptions are also possible with offers so that would need to be part of this flow (could just be a toggle to say making this a recurring request or payment).

This issue would touch on quite a few pages. 

When I get to revising the requesting page I will probably change the `Withdrawal request` heading to `Pull requests` and include LNURL-withdraw (already there) and Offers as a start.

**Comment by sbddesign** [comment]
> I can scan the offer, and rather then them serve me an invoice (how standard invoices are served) I can send them an invoice from my node that they pay.

This is all I was trying to convey -- that `send_invoice` happens in response to an offer.

> When I get to revising the requesting page I will probably change the Withdrawal request heading to Pull requests and include LNURL-withdraw (already there) and Offers as a start.

Pull request sounds a little obtuse, IMHO. "Withdraw" is more in line with traditional finance language I am accustomed to. Do you typically use "pull" in conversations around money, finance, etc?

**Comment by Bosch-0** [comment]
> This is all I was trying to convey -- that send_invoice happens in response to an offer.

Yeah I was just laying out a scenario for readers of this issue to better explain things

> Do you typically use "pull" in conversations around money, finance, etc?

Yeah I agree, withdraw is a little less jargony than pull.

**Comment by Bosch-0** [comment]
This could be a fun case study to explore this UX. Probably not ready to be apart of a reference design

**Comment by pavlenex** [comment]
@Bosch-0 It seems we addressed some of these here https://bitcoin.design/guide/daily-spending-wallet/requesting/, this can potentially be something we add once we create merchant section, talk a bit more specifically in context of a refund? #864
